components/BottomNavigation/src/private/MDCBottomNavigationSystemDialogView.m

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBottomNavigationSystemDialogView.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+
+static const CGFloat kCornerRadius = 10;
+static const CGFloat kMargins = 10;
+
+static UIVisualEffectView *MDCInitializeCompatibleBlurView() {
+  if (@available(iOS 10, *)) {
+    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleProminent];
+    UIVisualEffectView *blurView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    return blurView;
+  }
+
+  UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
+  return [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+}
+
+@interface MDCBottomNavigationSystemDialogView ()
+
+/**
+ * The background blur view. This property is nil if on instantiation reduced transparency was
+ * enabled.
+ */
+@property(nonatomic, nullable) UIVisualEffectView *blurView;
+
+@end
+
+@implementation MDCBottomNavigationSystemDialogView
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonInitMDCBottomNavigationSystemDialogView];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonInitMDCBottomNavigationSystemDialogView];
+  }
+
+  return self;
+}
+
+- (void)commonInitMDCBottomNavigationSystemDialogView {
```

Comment:
Nit/naming: Please name this `commonMDCBottomNavigationSystemDialogViewInit` to match the general pattern of the code base.

Reply:
Done.